### PR TITLE
Adding ecpool testing for rbd_clonev2 and rbd_faster_exports module

### PR DESCRIPTION
**Pull request by manasagowri**
Signed-off-by: mgowri <mgowri@redhat.com>

# Description

Adding logic to include both ecpool and replicated pool testing for the modules rbd_clonev2 and rbd_faster_exports

Success log for rbd_clonev2 - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-kox4j/clones_creation_with_v2clone_format_0.log

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/rbd/rbd_clonev2.py [hunk 1]
```diff
@@ -1,10 +1,72 @@
...
+        if rbd.trash_exist(pool, image):
+            log.error("Images are found in Trash")
+            return 1
+        return 0
+
+    except RbdBaseException as error:
+        print(error.message)
+        return 1
+
+    finally:
```
Comment (sunilangadi2) [inline tests/rbd/rbd_clonev2.py:66]: can we add this condition here `if not kw.get("config").get("do_not_cleanup_pool"):` ?

File: tests/rbd/rbd_clonev2.py [hunk 2]
```diff
@@ -33,37 +95,22 @@ def run(**kw):
...
-    clone = rbd.random_string()
-    size = "1G"
+    if kw.get("config"):
+        kw["config"]["ec-pool-k-m"] = "use_default"
+    else:
+        kw = {"config": {"ec-pool-k-m": "use_default"}}
+    rbd_obj = initial_rbd_config(**kw)
+    rc = 1
+    if rbd_obj:
+        rc = rbd_clonev2(rbd_obj.get("rbd_reppool"), "rep_pool_config", **kw)
```
Comment (sunilangadi2) [inline tests/rbd/rbd_clonev2.py:109]: can we have some info messages saying test running on **EC pools** and **Replicated pools**?

**Review by sunilangadi2 — COMMENTED** [review]
@manasagowri Does these changes not needed any **test conf** changes in EC config?

File: tests/rbd/rbd_clonev2.py (hunk 1 above)
Comment (manasagowri) [reply tests/rbd/rbd_clonev2.py:66]: Done.

File: tests/rbd/rbd_clonev2.py (hunk 2 above)
Comment (manasagowri) [reply tests/rbd/rbd_clonev2.py:109]: Done.

**Comment by manasagowri** [comment]
> @manasagowri Does these changes not needed any **test conf** changes in EC config?

@sunilangadi2 No changes required in test conf since we are using all default values.

**Review by sunilangadi2 — DISMISSED** [review]
Changes look good.

**Review by Manohar-Murthy — APPROVED** [review]
lgtm.

**Pull request merged**